fin_control/src/fin_control.cpp

```diff
@@ -261,107 +260,58 @@ void FinControl::handleSetAngles(const fin_control::SetAngles::ConstPtr &msg)
     return;
   }
 
-  // fin1
-  angle_plus_offset =
-      ctrlFinScaleFactor * (msg->f1_angle_in_radians + ctrlFinOffset);
-
-  if (!(myWorkBench.addBulkWriteParam(1, "Goal_Position",
-                                      myWorkBench.convertRadian2Value(1, angle_plus_offset), &log)))
-  {
-    ROS_ERROR("Could not add bulk write param 1 %s", log);
-    return;
-  }
-
-  // fin2
-  angle_plus_offset =
-      ctrlFinScaleFactor * (msg->f2_angle_in_radians + ctrlFinOffset);
-
-  if (!(myWorkBench.addBulkWriteParam(2, "Goal_Position",
-                                      myWorkBench.convertRadian2Value(2, angle_plus_offset), &log)))
-  {
-    ROS_ERROR("Could not add bulk write param 2 %s", log);
-    return;
-  }
-
-  // fin3
-  angle_plus_offset =
-      ctrlFinScaleFactor * (msg->f3_angle_in_radians + ctrlFinOffset);
-
-  if (!(myWorkBench.addBulkWriteParam(3, "Goal_Position",
-                                      myWorkBench.convertRadian2Value(3, angle_plus_offset), &log)))
-  {
-    ROS_ERROR("Could not add bulk write param 3 %s", log);
-    return;
-  }
-
-  // fin4
-  angle_plus_offset =
-      ctrlFinScaleFactor * (msg->f4_angle_in_radians + ctrlFinOffset);
-
-  if (!(myWorkBench.addBulkWriteParam(4, "Goal_Position",
-                                      myWorkBench.convertRadian2Value(4, angle_plus_offset), &log)))
-  {
-    ROS_ERROR("Could not add bulk write param 4 %s", log);
-    return;
-  }
-
-  if (!myWorkBench.bulkWrite(&log))
-  {
-    ROS_ERROR("Could not bulk write %s", log);
-  }
-
-  if (currentLoggingEnabled)
+  // check for max angle the fins can mechanically handle
+  for (uint8_t i = 0; i < 4; i++)
   {
-    int32_t cdata1, cdata2, cdata3, cdata4;
-    cdata1 = cdata2 = cdata3 = cdata4 = 0;
+    angle_plus_offset = ctrlFinScaleFactor * (msg->fin_angle_in_radians[i] + ctrlFinOffset);
 
-    if (!myWorkBench.itemRead(1, "Present_Current", &cdata1, &log))
-    {
-      ROS_ERROR("Could Read F1 Current %s", log);
-    }
-    else if (!myWorkBench.itemRead(2, "Present_Current", &cdata2, &log))
+    if (!(myWorkBench.addBulkWriteParam(i + 1, "Goal_Position",
```

Comment:
@patapufate nit: consider having a `const int id = i + 1;` variable instead of replicating `i + 1`.

Reply:
fixed in 772413b